src/main/scala/chisel3/util/BoringUtils.scala

```diff
@@ -0,0 +1,59 @@
+// See LICENSE for license details.
+
+package chisel3.util
+
+import chisel3._
+import chisel3.experimental._
+import chisel3.internal.{InstanceId, NamedComponent}
+import firrtl.transforms.DontTouchAnnotation
+import firrtl.passes.wiring.{WiringTransform, SourceAnnotation, SinkAnnotation}
+
+/** Utilities for generating synthesizeable cross module references.
+  *
+  * @example {{{
+  * class ModuleA extends Module with BoringUtils {
+  *   val a = Reg(Bool())
+  *   addSource(a, "unique_identifier")
+  * }
+  * class ModuleB extends Module with BoringUtils {
+  *   val b = Wire(Bool())
+  *   addSink(b, "unique_identifier")
+  * }
+  * class ModuleC extends Module with BoringUtils {
+  *   val c = Wire(Bool())
+  *   addSink(c, "unique_identifier")
+  * }
+  * }}}
+  */
+trait BoringUtils extends BaseModule {
```

Comment:
Since `annotate` is now an object with an apply method, is it necessary to make this be a trait that has to be mixed in by a given module?